src/main/scala/org/hammerlab/guacamole/reads/ReadPair.scala

```diff
@@ -0,0 +1,41 @@
+package org.hammerlab.guacamole.reads
+
+import htsjdk.samtools.SAMRecord
+
+/**
+ * *
+ * @param read Unmapped or MappedRead base read
+ * @param isFirstInPair Whether the read is earlier that the the mate read
+ * @param mateAlignmentProperties Alignment location of the mate if it the mate is aligned
+ * @tparam T UnmappedRead or MappedRead
+ */
+case class ReadPair[+T <: Read](read: T,
```

Comment:
This class name seems confusing to me since I'd expect a `ReadPair` object to contain two reads. What does this class represent? I'd suggest a header line in the doc summarizing what this is, and potentially renaming it to something clearer (e.g. `PairedRead`?)